api/setting.go

```diff
@@ -98,6 +151,31 @@ type (
 	}
 )
 
+// IsPinned returns true if the pin is enabled and the value is greater than 0.
+func (p Pin) IsPinned() bool {
+	return p.Pinned && p.Value > 0
+}
+
+// Validate returns an error if the price pin settings are not considered valid.
+func (pps PricePinSettings) Validate(ctx context.Context) error {
+	if !pps.Enabled {
+		return nil
+	}
+	if pps.ForexEndpointURL == "" {
+		return fmt.Errorf("price pin settings must have a forex endpoint URL")
+	}
+	if pps.Currency == "" {
+		return fmt.Errorf("price pin settings must have a currency")
+	}
+	if pps.Threshold <= 0 || pps.Threshold >= 1 {
+		return fmt.Errorf("price pin settings must have a threshold between 0 and 1")
+	}
+	if _, err := ibus.NewForexClient(pps.ForexEndpointURL).SiacoinExchangeRate(ctx, pps.Currency); err != nil {
+		return fmt.Errorf("couldn't update price pinning settings, forex API unreachable,error: %v", err)
+	}
+	return nil
```

Comment:
Might be a good spot to try the client. That would implicitly validate that the URL is sane and the currency as well.

Reply:
That's exactly what I'm doing?